### PR TITLE
Add wheel/python-version/python-implementation/status badage for PyPI.

### PR DESCRIPTION
**Pull request by tonyseek**
Please review it. Thanks.

Related Issues: https://github.com/badges/pypipins/issues/39 https://github.com/badges/pypipins/issues/37

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -511,6 +511,22 @@ <h3 id="miscellaneous"> Miscellaneous </h3>
...
+    <td><code>https://img.shields.io/pypi/py_versions/Django.svg</code></td>
+  </tr>
+  <tr><th data-keywords='python'> PyPI: </th>
+    <td><img src='https://img.shields.io/pypi/implementation/Django.svg' alt=''/></td>
+    <td><code>https://img.shields.io/pypi/implementation/Django.svg</code></td>
+  </tr>
+  <tr><th data-keywords='python'> PyPI: </th>
+    <td><img src='https://img.shields.io/pypi/status/Django.svg' alt=''/></td>
+    <td><code>https://img.shields.io/pypi/status/Django.svg</code></td>
+  </tr>
```
Comment (espadrine) [inline index.html]: `index.html` is auto-generated. Apply modifications to try.html instead.

**Comment by tonyseek** [comment]
@espadrine I modified the `try.html` and reverted the `index.html`.

File: server.js [hunk 2]
```diff
@@ -1481,6 +1481,73 @@ cache(function(data, match, sendBadge, request) {
...
+          versions.push('not found');
+        }
+        badgeData.text[0] = 'python';
+        badgeData.text[1] = versions.sort().join(', ');
+        badgeData.colorscheme = 'blue';
+        sendBadge(format, badgeData);
+      } else if (info == 'implementation') {
+        var implementations = [];
+        var pattern = /^Programming Language \:\: Python \:\: Implementation \:\: (\S+)$/;
+        for (var i in data.info.classifiers) {
```
Comment (espadrine) [inline server.js]: Could you use a regular loop here, please?
```js
for (var i = 0; i < data.info.classifiers.length; i++) {
…
}
```

(Yes, I agree, JS is ugly compared to Python ☺.)

File: server.js [hunk 3]
```diff
@@ -1481,6 +1481,73 @@ cache(function(data, match, sendBadge, request) {
...
+          }
+        }
+        badgeData.text[0] = 'wheel';
+        badgeData.text[1] = hasWheel ? 'yes' : 'no';
+        badgeData.colorscheme = hasWheel ? 'brightgreen' : 'red';
+        sendBadge(format, badgeData);
+      } else if (info == 'py_versions') {
+        var versions = [];
+        var pattern = /^Programming Language \:\: Python \:\: (\d\.\d)$/;
+        for (var i in data.info.classifiers) {
```
Comment (espadrine) [inline server.js]: Could you use a regular loop here too?

File: server.js [hunk 4]
```diff
@@ -1481,6 +1481,73 @@ cache(function(data, match, sendBadge, request) {
...
+        badgeData.text[1] = implementations.sort().join(', ');
+        badgeData.colorscheme = 'blue';
+        sendBadge(format, badgeData);
+      } else if (info == 'status') {
+        var pattern = /^Development Status \:\: ([1-7]) - (\S+)$/;
+        var statusColors = {
+            '1': 'red', '2': 'red', '3': 'red', '4': 'yellow',
+            '5': 'brightgreen', '6': 'brightgreen', '7': 'red'};
+        var statusCode = '1', statusText = 'unknown';
+        for (var i in data.info.classifiers) {
```
Comment (espadrine) [inline server.js]: Oh, and here as well.

File: server.js (hunk 2 above)
Comment (tonyseek) [inline server.js]: Okay.

File: server.js [hunk 5]
```diff
@@ -1481,6 +1481,73 @@ cache(function(data, match, sendBadge, request) {
           badgeData.colorscheme = 'blue';
         }
         sendBadge(format, badgeData);
+      } else if (info == 'wheel') {
+        var releases = data.releases[data.info.version];
+        var hasWheel = false;
+        for (var i in releases) {
```
Comment (espadrine) [inline server.js]: Finally, there.

**Comment by espadrine** [comment]
Thanks a lot for contributing! I hope all is well on @kura's side.

(As a tiny extra, it would be fantastic if you'd use `===` instead of `==`.)

File: try.html [hunk 6]
```diff
@@ -518,6 +518,22 @@ <h3 id="miscellaneous"> Miscellaneous </h3>
     <td><img src='/pypi/l/Django.svg' alt=''/></td>
     <td><code>https://img.shields.io/pypi/l/Django.svg</code></td>
   </tr>
+  <tr><th data-keywords='python'> PyPI: </th>
+    <td><img src='/pypi/wheel/Django.svg' alt=''/></td>
+    <td><code>https://img.shields.io/pypi/wheel/Django.svg</code></td>
+  </tr>
+  <tr><th data-keywords='python'> PyPI: </th>
+    <td><img src='/pypi/py_versions/Django.svg' alt=''/></td>
```
Comment (espadrine) [inline try.html]: You are using `pyversions` above. Which do you prefer?

Comment (tonyseek) [inline try.html]: I'm sorry. I forgot it. It will be fixed :-)

**Comment by tonyseek** [comment]
@espadrine I pushed new commits.

**Comment by espadrine** [comment]
10/10 will merge and prod.

**Comment by tonyseek** [comment]
Thank you!

**Comment by espadrine** [comment]
It's live. Try it, tell me if there is a bug we didn't anticipate; I'll update the website in 24h to invite everyone to use it.

**Comment by tonyseek** [comment]
:+1:

**Comment by josuebrunel** [comment]
Hello guys, 
It seems like ***py_versions*** isn't working

https://img.shields.io/pypi/py_versions/Django.svg

For my project

https://img.shields.io/pypi/py_versions/myql.svg

[PyPI](https://pypi.python.org/pypi/myql)

Thanks

**Comment by tonyseek** [comment]
@josuebrunel `https://img.shields.io/pypi/pyversions/Django.svg` (without the underline) works here.

**Comment by josuebrunel** [comment]
@tonyseek thanks a lot. Great job :+1:

**Comment by tonyseek** [comment]
https://github.com/tonyseek/shields/commit/dee58b43c459eb8cc46f5bb780340657996c156a is a temporary commit :-)

**Comment by xflr6** [comment]
Great work, thanks.
Any chance to add a replacement for the [format badge](https://github.com/badges/pypipins/blob/master/docs/source/format.rst) (somewhat more informative than wheel: yes/no)?
And maybe make the version badge green (red looking like a warning)?

**Comment by espadrine** [comment]
@xflr6 yes, that would be great! What did it look like? `format | wheel` etc.?

**Comment by xflr6** [comment]
Yes, the best available format from [`wheel`, `source`, or `edge`](https://github.com/badges/pypipins/blob/master/shields/shields.py#L194-203) (red, yellow, green).

**Comment by tonyseek** [comment]
In my opinion keeping `wheel` be standalone may be better, because it is a [encouraged](http://pythonwheels.com) format rather than `egg`(outdated) and `source`(almost all packages have it). :whale:

**Comment by espadrine** [comment]
Further information on how to detect wheel / egg: https://github.com/kura/yarg/blob/master/yarg/package.py#L241

**Comment by espadrine** [comment]
@xflr6 this should work: https://img.shields.io/pypi/format/graphviz.svg

As usual, I'll update the website in 24h.

**Comment by xflr6** [comment]
Great, thanks a lot!

As for the color of the version badge, I just realized that mine are red because I conservatively keep my packages  in the `0.x` versions. I see the point, but then I would like to second #349 for adding an optional parameter to override the color (at least for the version badge so we have full backwards-compatibility with pypip).

**Comment by espadrine** [comment]
@xflr6 I wish to keep backwards-compatibility with pypip. Can you specify the difference between the current state and pypip?

**Comment by xflr6** [comment]
Sorry this did not come through: The version badge was green regardless of the version (so a color-overriding parameter would make it possible to make the badge look exactly like before).

**Pull request merged**